### PR TITLE
Add AddCommentToMethod recipe

### PR DESCRIPTION
**Pull request by zoewangg**
## Motivation and Context
There are some methods that are removed in Java SDK v2 and for those changes, we'll not update the code and instead, just add a comment providing suggestions.

## Modifications
Add AddCommentToMethod recipe

## Testing
Unit tests added

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md) document
- [ ] Local run of `mvn install` succeeds
- [ ] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] I have added a changelog entry. Adding a new entry must be accomplished by running the `scripts/new-change` script and following the instructions. Commit the new file created by the script in `.changes/next-release` with your changes.
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: migration-tool/src/main/java/software/amazon/awssdk/migration/internal/recipe/AddCommentToMethod.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new Visitor(methodPattern, comment);
+    }
+
+    private static final class Visitor extends JavaIsoVisitor<ExecutionContext> {
+        private final MethodMatcher methodMatcher;
+        private final String comment;
+
+        Visitor(String methodPattern, String comment) {
+            this.methodMatcher = new MethodMatcher(methodPattern, false);
```
Comment (davidh44) [inline migration-tool/src/main/java/software/amazon/awssdk/migration/internal/recipe/AddCommentToMethod.java:73]: Are there any cases we would want the matchOverrides flag to be true?

File: migration-tool/src/main/java/software/amazon/awssdk/migration/internal/recipe/AddCommentToMethod.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    @Option(displayName = "Comment",
+        description = "A comment to add to this method.",
+        example = "This method is not supported in AWS SDK for Java v2.")
+    private final String comment;
+
+    @JsonCreator
+    public AddCommentToMethod(@JsonProperty("methodPattern") String methodPattern,
+                              @JsonProperty("comment") String comment) {
+        this.methodPattern = methodPattern;
+        this.comment = comment;
```
Comment (davidh44) [inline migration-tool/src/main/java/software/amazon/awssdk/migration/internal/recipe/AddCommentToMethod.java:50]: Should we begin each comment making it clear it is from the migration tool? e.g., have a constant String to distinguish when there are existing comments

File: migration-tool/src/test/java/software/amazon/awssdk/migration/recipe/AddCommentToMethodTest.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                + "    }\n"
+                + "}\n",
+                "import com.amazonaws.ClientConfiguration;\n"
+                + "\n"
+                + "public class Example {\n"
+                + "    \n"
+                + "    void test() {\n"
+                + "        ClientConfiguration clientConfiguration = new ClientConfiguration();\n"
+                + "        // Existing comment \n"
+                + "        /*existing comment*/ /*~~(a comment)~~>*/clientConfiguration.setCacheResponseMetadata(false);\n"
```
Comment (davidh44) [inline migration-tool/src/test/java/software/amazon/awssdk/migration/recipe/AddCommentToMethodTest.java:84]: Does the `/* comment */` get rendered on its own line? Don't see a `\n` as above with the `// comment`

Comment (zoewangg) [reply migration-tool/src/test/java/software/amazon/awssdk/migration/recipe/AddCommentToMethodTest.java:84]: It's rendered on the same line of the code. I followed the same style that openrewrite used for their recipes

File: migration-tool/src/main/java/software/amazon/awssdk/migration/internal/recipe/AddCommentToMethod.java (hunk 2 above)
Comment (zoewangg) [reply migration-tool/src/main/java/software/amazon/awssdk/migration/internal/recipe/AddCommentToMethod.java:50]: Yeah, good idea!

File: migration-tool/src/main/java/software/amazon/awssdk/migration/internal/recipe/AddCommentToMethod.java (hunk 1 above)
Comment (zoewangg) [reply migration-tool/src/main/java/software/amazon/awssdk/migration/internal/recipe/AddCommentToMethod.java:73]: Probably not, I can't think of a scenario where we want to match overrides for our use-case.

File: migration-tool/src/test/java/software/amazon/awssdk/migration/recipe/AddCommentToMethodTest.java (hunk 3 above)
Comment (davidh44) [reply migration-tool/src/test/java/software/amazon/awssdk/migration/recipe/AddCommentToMethodTest.java:84]: Got it, so this is just for test purposes. I assume we'd want the added comment on a separate line than the existing comment?

**Review by davidh44 — APPROVED** [review]

File: migration-tool/src/test/java/software/amazon/awssdk/migration/recipe/AddCommentToMethodTest.java (hunk 3 above)
Comment (zoewangg) [reply migration-tool/src/test/java/software/amazon/awssdk/migration/recipe/AddCommentToMethodTest.java]: Yeah, ideally, I tried to do that but it caused some other formatting issues; it's surprising hard to get the indent and formatting right after we add a new comment, so I just used one of openrewrite classes directly to generate the comment, which happens to be on the same line.

**Pull request merged**